nimble/controller/src/ble_ll_conn.c

```diff
@@ -3612,22 +3611,33 @@ ble_ll_conn_rx_isr_end(uint8_t *rxbuf, struct ble_mbuf_hdr *rxhdr)
     uint32_t add_usecs;
     struct os_mbuf *txpdu;
     struct ble_ll_conn_sm *connsm;
-    struct os_mbuf *rxpdu;
+    struct os_mbuf *rxpdu = NULL;
     struct ble_mbuf_hdr *txhdr;
     int rx_phy_mode;
+    bool alloc_rxpdu = true;
 
     /* Retrieve the header and payload length */
     hdr_byte = rxbuf[0];
     rx_pyld_len = rxbuf[1];
 
+    /*
+     * No need to alloc rxpdu for packets with invalid CRC, we would throw them
+     * away instantly from LL anyway.
+     */
+    if (!BLE_MBUF_HDR_CRC_OK(rxhdr)) {
```

Comment:
Even we throw away packet in LL, it does not mean that this packet is not important. e.g. when initiating connection using ext commands it is import for us to handle invalid AUX_CONNECT_RSP to do some cleaning and restart scanning. Maybe this is a good idea to handle this way but we need more work on that and in the same time remove all the CRC checks in higher layer, if we decide to drop such packets in here.

Reply:
ah you are right, we are in "ble_ll_conn_rx_isr_end" which when connection is up. So indeed it should be fine.